### PR TITLE
grpcio: Changes backported from TF 2.15 branch

### PR DESCRIPTION
**Pull request by smuzaffar**
FYI @iarspider

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@iarspider , can you please try building it locally. I think grpcio is not linking absl libs as it is looking for those libs under [/usr/lib](https://github.com/grpc/grpc/blob/master/setup.py#L363-L366)

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
@iarspider , looks like you are not applying the newly added patch [grpcio-absl-path.patch](https://github.com/cms-sw/cmsdist/pull/9015/files#diff-1cecd6b9b18d4f0a77a7650dacaa2f9df57f35f063f1859cc2e7e6ef68623e34) . log file https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c095c1/37454/externals/py3-grpcio/1.60.1-e281228763f985e50a5c701b73938cd9/  should that it is still not linking our absl libs

File: grpcio-absl-path.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/setup.py	2024-02-14 12:21:26.788212443 +0100
++++ b/setup.py	2024-02-14 14:00:46.036665638 +0100
+@@ -378,7 +378,7 @@
+     EXTENSION_LIBRARIES += ("re2",)
+ if BUILD_WITH_SYSTEM_ABSL:
+     EXTENSION_LIBRARIES += tuple(
+-        lib.stem[3:] for lib in pathlib.Path("/usr").glob("lib*/libabsl_*.so")
++        lib.stem[3:] for lib in pathlib.Path(os.getenv("ABSEIL_CPP_ROOT")).glob("lib*/libabsl_*.so")
```
Comment (smuzaffar) [inline grpcio-absl-path.patch:8]: @iarspider  , is `lib.stem[3:]`  correct for our path?

Comment (iarspider) [reply grpcio-absl-path.patch:8]: [Yes](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem): `stem` is "the final path component, without its suffix". So this line strips "lib" prefix and last suffix (extension)

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
@iarspider , we also have https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_14_1_X/master/c-ares.spec and I see grpcio is using the one from its thrid-party area. Can you configure it to use ours?

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**